lib/database_config.py

```diff
@@ -0,0 +1,362 @@
+        has_extra = "extra" in match.groupdict() and match.group("extra")
+        if has_extra:
+            extra = match.group("extra").lstrip("?")
+            jdbc_params = parse_qs(extra)
+
+        # defaults
+        if database_type == "PostgreSQL":
+            jdbc_params.update({"tcpKeepAlive": "true"})
+
+        extra_url_params_str = os.getenv("DATABASE_CONNECTION_PARAMS", "{}")
+        if extra_url_params_str is not None:
+            try:
+                extra_url_params = json.loads(extra_url_params_str)
+                jdbc_params.update(extra_url_params)
+            except Exception:
+                logger.warning(
+                    "Invalid JSON string for DATABASE_CONNECTION_PARAMS"
+                )
+
+        # generate jdbc_url, might be None
+        jdbc_url = self.get_jdbc_strings(self.url, match, config, jdbc_params)
+        if jdbc_url is not None:
+            logger.debug("Setting JDBC url: {}".format(jdbc_url))
+            config.update({"DatabaseJdbcUrl": jdbc_url})
+
+        if "sslmode" in jdbc_params:
+            sslmode = jdbc_params["sslmode"]
+            if sslmode and sslmode[0] == "require":
+                config.update({"DatabaseUseSsl": True})
+
+        if database_type_input == "mysql":
+            config.update(
+                {
+                    "ConnectionPoolingNumTestsPerEvictionRun": 50,
+                    "ConnectionPoolingSoftMinEvictableIdleTimeMillis": 10000,
+                    "ConnectionPoolingTimeBetweenEvictionRunsMillis": 10000,
+                }
+            )
+
+        return config
+
+    def get_jdbc_strings(self, url, match, config, jdbc_params):
+        # JDBC strings might be different from connection uri strings retrieved from the VCAP
+        # For supported/tested situations we'll create a JDBC string based on
+        # * url (from VCAP or DATABASE_URL)
+        # * config (extracted information from url)
+        # * jdbc_params (from DATABASE_URL or DATABASE_CONNECTION_PARAMS)
+        #
+        # if given url is a JDBC string this will be returned
+        #
+
+        # return unmodified jdbc string
+        if url.startswith("jdbc:"):
+            return url
+
+        if len(jdbc_params) > 0:
+            extra_jdbc_params = "?{}".format(urlencode(jdbc_params, True))
+        else:
+            extra_jdbc_params = ""
+
+        if config["DatabaseType"] == "PostgreSQL":
+            jdbc_url = "jdbc:postgresql://{}/{}{}".format(
+                config["DatabaseHost"],
+                config["DatabaseName"],
+                extra_jdbc_params,
+            )
+            return jdbc_url
+
+
+class SapHanaDatabaseConfiguration(DatabaseConfiguration):
+
+    database_type = "SAPHANA"
+
+    def __init__(self, credentials):
+        super().__init__()
+        logger.info("Detected SAP Hana configuration.")
+        self.credentials = credentials
+
+    def parse_configuration(self):
+        url = self.credentials["url"]
```

Comment:
You can make this simpler by just reusing the credentials dict as opposed to assigning all of these params to vars